Source/dom/components/Layer.js

```diff
@@ -63,36 +63,25 @@ export class Layer extends WrappedObject {
     return this
   }
 
-  /**
-   * Convert a rectangle in the coordinates that this layer uses to absolute (page) coordinates.
-   *
-   * @param {Rectangle} rect The rectangle to convert.
-   * @return {Rectangle} The converted rectangle expressed in page coordinates.
-   */
+  // @deprecated
   localRectToPageRect(rect) {
-    const _rect = this._object.convertRectToAbsoluteCoordinates(rect.asCGRect())
-    return new Rectangle(
-      _rect.origin.x,
-      _rect.origin.y,
-      _rect.size.width,
-      _rect.size.height
+    console.warn(
+      'Layer.layerRectToPageRect(rect) is deprecated. Use rect.changeBasis({ from: layer }) instead'
     )
+    return rect.changeBasis({
+      from: this,
+    })
   }
 
-  /**
-   * Convert a rectangle in the coordinates that this layer uses to it's parent's coordinates.
-   *
-   * @param {Rectangle} rect The rectangle to convert.
-   * @return {Rectangle} The converted rectangle expressed in the coordinate system of the parent layer.
-   */
+  // @deprecated
   localRectToParentRect(rect) {
```

Comment:
api philosophy question, as I know you've mentioned this before - is the idea to reduce the footprint of the JS api as small as possible, and leave convenience methods to user-land? In this case, it would still be cool to have `layer.parentRect` and `layer.pageRect` as that probably covers a lot of cases on what people will be trying to do, instead of them having to go and find the concepts of `basis` and get their head around that.

Reply:
I added some examples to about getting the `parentRect` and `pageRect`. Hopefully, that will make it easier.

Otherwise, feel free to create a `sketch-geometry` package and I'll be happy to link to it :)

Reply:
yep the examples are there, was more a general question as we seem to be against these sort of convenience methods.